### DISCUSSION TITLE
Player teleporting functionality

### DISCUSSION THREAD

**Discussion by vurvdev** [wiremod/wire#2676] [Suggestions]
Player teleportation is probably one of the most common use cases for external E2 extensions.

Need that in base wiremod, likely through a convar to allow access on players.

**Comment by vurvdev** [comment]
Could be a wider convar that just allows all of the entity/propcore functions to work on players, granted you have the provided prop protection perms to do so.

**Reply by Hazeofdream** [reply]
can't you bypass that requirement by using a wire teleporter and target finder? i get the restrictions but if you can do it with gates then it has no reason to be restricted on e2.

**Reply by vurvdev** [reply]
It does. E2 can't have functions restricted easily, and it's much less obvious who is the one minging. The teleporter can just be blocked with prop protection.

**Reply by DecVes** [reply]
very nice but not only just tp function, I do agree with your statement of allowing players to use basically anything on them with the right permissions depending on what it is. If you want to you **_could_** make it so you can affect other players but i have no clue how that would work

**Comment by Denneisk** [comment]
Convar is probably the easiest but I feel like there's no pleasant solution. Would be nice to have the functionality in its own extension, because having player manipulation in "propcore" is a bit misleading, but still use the `setPos` name, but players are just entities and making them a type is bloat.
My ideal solution would be make E2 OOP and use polymorphism.

**Comment by DecVes** [comment]
> Player teleportation is probably one of the most common use cases for external E2 extensions.
> 
> Need that in base wiremod, likely through a convar to allow access on players.

as stated in one of my comments on this page you could add other things instead of just teleportation functions  Read my other comment to see what i mean

**Comment by bjcscat** [comment]
I don't imagine this being used in places where propcore (or equivalent) isn't used so a convar to allow the propcore setPos to work on players is acceptable. Although having it behind a convar for its own function (e:teleportPlayer(v)) could account for as of yet unknown usecases. Perhaps a convar could contain information on how it should work such as:
* 0 - Disabled
* 1 - Admins only
* 2 - VGUI Prompt (needs further discussion)
* 3 - All
Teleportation should inform players about what E2 teleported you with a chat message or similar.

**Reply by vurvdev** [reply]
It would be up to prop protection, and vgui prompt wouldn't be an option just cause of how involved it'd be.

The convar would just be there for the server's sake, for darkrp or servers that don't want players teleporting around.

**Reply by bjcscat** [reply]
I don't think most prop protection addons really involve themselves with the teleportation of players.

**Reply by vurvdev** [reply]
They should give you permissions to toolgun a player. That's what starfall checks

**Reply by Divran** [reply]
One of the options should be "only yourself"

**Comment by bjcscat** [comment]
I think that teleporting players should be the function of a different entity, not E2. A convar could be added to allow for propcore functions to be used on players that the E2 owner has toolgun perms on but all other cases should be handled through another entity.
